Source/WebCore/page/FrameView.cpp

```diff
@@ -2241,8 +2244,11 @@ bool FrameView::scrollToFragment(const URL& url)
                 document->fragmentHighlightRegister().addAnnotationHighlightWithRange(StaticRange::create(range));
             
             if (highlightRanges.size()) {
-                TemporarySelectionChange selectionChange(document, { highlightRanges.first() }, { TemporarySelectionOption::DelegateMainFrameScroll, TemporarySelectionOption::SmoothScroll, TemporarySelectionOption::RevealSelectionBounds });
-                // FIXME: add a textIndicator after the scroll has completed.
+                auto range = highlightRanges.first();
+                TemporarySelectionChange selectionChange(document, { range }, { TemporarySelectionOption::DelegateMainFrameScroll, TemporarySelectionOption::RevealSelectionBounds, TemporarySelectionOption::UserTriggered });
+                m_pendingTextFragmentIndicatorRange = range;
+                m_pendingTextFragmentIndicatorText = plainText(range);
+                m_delayedTextFragmentIndicatorTimer.startOneShot(100_ms);
```

Comment:
100ms seems really slow. Humans can perceive that. Do we really need to wait that long?

Reply:
For example, HTML5 event loop defines when we fire scroll events:
https://html.spec.whatwg.org/multipage/webappapis.html#update-the-rendering

Perhaps that is more appropriate time to wait for?

Reply:
This is a human-factors timer (load+scroll -> take a beat -> flash), not here for a technical reason, so it's good that it's a perceptible time, IMO :)

Reply:
I tried 100ms and 250ms, I felt that 100 was better, but HI is going to be the real decision maker here, I didn't want to spend too much time on a number that was very likely to be overridden.
And yes, it should be human-perceivable, as Tim said. We _want_ to make sure that it can be seen.